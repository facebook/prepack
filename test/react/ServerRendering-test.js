/**
 * Copyright (c) 2017-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

/* @flow */

const fs = require("fs");
const setupReactTests = require("./setupReactTests");
const { runTest } = setupReactTests();

/* eslint-disable no-undef */
const { it } = global;

it("Hacker News app", () => {
  let data = JSON.parse(fs.readFileSync(__dirname + "/ServerRendering/hacker-news.json").toString());
  runTest(__dirname + "/ServerRendering/hacker-news.js", { data });
});

it("PE Functional Components benchmark", () => {
  runTest(__dirname + "/ServerRendering/pe-functional-benchmark.js");
});
