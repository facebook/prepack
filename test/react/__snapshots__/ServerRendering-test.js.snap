// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hacker News app: (JSX => JSX) 1`] = `
ReactStatistics {
  "componentsEvaluated": 0,
  "evaluatedRootNodes": Array [],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 0,
}
`;

exports[`Hacker News app: (JSX => createElement) 1`] = `
ReactStatistics {
  "componentsEvaluated": 0,
  "evaluatedRootNodes": Array [],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 0,
}
`;

exports[`Hacker News app: (createElement => JSX) 1`] = `
ReactStatistics {
  "componentsEvaluated": 0,
  "evaluatedRootNodes": Array [],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 0,
}
`;

exports[`Hacker News app: (createElement => createElement) 1`] = `
ReactStatistics {
  "componentsEvaluated": 0,
  "evaluatedRootNodes": Array [],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 0,
}
`;

exports[`PE Functional Components benchmark: (JSX => JSX) 1`] = `
ReactStatistics {
  "componentsEvaluated": 0,
  "evaluatedRootNodes": Array [],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 0,
}
`;

exports[`PE Functional Components benchmark: (JSX => createElement) 1`] = `
ReactStatistics {
  "componentsEvaluated": 0,
  "evaluatedRootNodes": Array [],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 0,
}
`;

exports[`PE Functional Components benchmark: (createElement => JSX) 1`] = `
ReactStatistics {
  "componentsEvaluated": 0,
  "evaluatedRootNodes": Array [],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 0,
}
`;

exports[`PE Functional Components benchmark: (createElement => createElement) 1`] = `
ReactStatistics {
  "componentsEvaluated": 0,
  "evaluatedRootNodes": Array [],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 0,
}
`;
